src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static Random random;
+    private ColorSupplier colorSupplier;
+
+    public FigureSupplier(Random random, ColorSupplier colorSupplier) {
+        this.random = random;
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        String color = colorSupplier.getRandomColor();
+        String[] figureTypes = {"Square", "Rectangle",
+                "Right Rectangle", "Circle", "Isosceles Trapezoid"};
+        int figureTypeIndex = random.nextInt(figureTypes.length);
+        String figureType = figureTypes[figureTypeIndex];
+        return createFigure(figureType, random, colorSupplier);
+    }
+
+    private static Figure createFigure(String figureType,
+                                       Random random,
+                                       ColorSupplier colorSupplier) {
+        Figure figure;
+        switch (figureType) {
+            case "Square":
+                double side = random.nextDouble() * 10 + 1;
+                figure = new Square(side, colorSupplier.getRandomColor());
```

Comment:
Let's create a `randomColor` (colorSupplier.getRandomColor()) variable before the `switch` and use it in all cases.